pkg/provider/azure_loadbalancer.go

```diff
@@ -1518,6 +1519,9 @@ func (az *Cloud) reconcileBackendPools(clusterName string, service *v1.Service,
 				for _, ipConf := range *bp.BackendIPConfigurations {
 					ipConfID := to.String(ipConf.ID)
 					nodeName, _, err := az.VMSet.GetNodeNameByIPConfigurationID(ipConfID)
+					if errors.Is(err, ErrorNotVmssInstance) { // Do nothing for the VMAS nodes.
```

Comment:
I'm a little confused why the new VMAS node would be typed VMSS?

Reply:
This is to address the type where the cluster "vmType" is set to "vmss"  to allow nodepools of different types (both VM and VMSS). See https://github.com/kubernetes-sigs/cloud-provider-azure/issues/338 for more context / previous discussion. I agree with you it's not ideal and confusing, which is why I'm working on a more long term fix to allow defining "mixed" clusters but this is needed in the meantime to unblock out-of-tree + CAPZ https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1216

Reply:
Yeah, I understand it now. But I think a better solution would be re-directing to VMAS.GetNodeNameByIPConfigurationID in VMSS.GetNodeNameByIPConfigurationID if the ipConfID matches the VMAS format. What do you think?

Reply:
Great suggestion. Let me test it locally to make sure it works.

Reply:
done, PTAL